src/platform/EFR32/Efr32PsaOperationalKeystore.cpp

```diff
@@ -0,0 +1,446 @@
+    VerifyOrReturnError(IsValidFabricIndex(fabricIndex), CHIP_ERROR_INVALID_FABRIC_INDEX);
+
+    // Remove pending keypair if we have it and the fabric ID matches
+    if ((mPendingKeypair != nullptr) && (fabricIndex == mPendingFabricIndex))
+    {
+        RevertPendingKeypair();
+    }
+
+    // Figure out which key ID we're looking for
+    EFR32OpaqueKeyId id = 0;
+    for (size_t i = 0; i < SL_MATTER_MAX_STORED_OP_KEYS; i++)
+    {
+        if (mKeyMapping[i] == fabricIndex)
+        {
+            id = i + kEFR32OpaqueKeyIdPersistentMin;
+
+            // Reset the key mapping since we'll be deleting this key
+            mKeyMapping[i] = kUndefinedFabricIndex;
+            break;
+        }
+    }
+
+    if (id < kEFR32OpaqueKeyIdPersistentMin || id > kEFR32OpaqueKeyIdPersistentMax)
+    {
+        // Fabric is not in the map, so assume it's gone already
+        return CHIP_NO_ERROR;
+    }
+
+    // Persist key map
+    CHIP_ERROR error = EFR32Config::WriteConfigValueBin(EFR32Config::kConfigKey_OpKeyMap, mKeyMapping, sizeof(mKeyMapping));
+    if (error != CHIP_NO_ERROR)
+    {
+        return error;
+    }
+
+    // Check if key is cached
+    EFR32OpaqueKeyId cachedId = mCachedKey->GetKeyId();
+
+    if (id == cachedId)
+    {
+        // Delete from persistent storage and unload
+        mCachedKey->Delete();
+        return CHIP_NO_ERROR;
+    }
+
+    // Load it for purposes of deletion
+    error = mCachedKey->Load(id);
+    if (error != CHIP_NO_ERROR && error != CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND)
+    {
+        return CHIP_ERROR_INTERNAL;
+    }
+
+    mCachedKey->Delete();
+
+    return CHIP_NO_ERROR;
+}
+
+void Efr32PsaOperationalKeystore::RevertPendingKeypair()
+{
+    if (mIsInitialized)
+    {
+        // Just delete the pending key from storage
+        ResetPendingKey();
+    }
+}
+
+CHIP_ERROR Efr32PsaOperationalKeystore::SignWithOpKeypair(FabricIndex fabricIndex, const ByteSpan & message,
+                                                          Crypto::P256ECDSASignature & outSignature) const
+{
+    VerifyOrReturnError(mIsInitialized, CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(IsValidFabricIndex(fabricIndex), CHIP_ERROR_INVALID_FABRIC_INDEX);
+
+    // Check to see whether the key is an activated pending key
+    if (mIsPendingKeypairActive && (fabricIndex == mPendingFabricIndex))
+    {
+        VerifyOrReturnError(mPendingKeypair != nullptr, CHIP_ERROR_INTERNAL);
+        return mPendingKeypair->ECDSA_sign_msg(message.data(), message.size(), outSignature);
+    }
+
+    // Figure out which key ID we're looking for
+    EFR32OpaqueKeyId id = 0;
+    for (size_t i = 0; i < SL_MATTER_MAX_STORED_OP_KEYS; i++)
+    {
+        if (mKeyMapping[i] == fabricIndex)
+        {
+            id = i + kEFR32OpaqueKeyIdPersistentMin;
+            break;
+        }
+    }
+
+    if (id < kEFR32OpaqueKeyIdPersistentMin || id > kEFR32OpaqueKeyIdPersistentMax)
```

Comment:
Same comments as above for the condition

Reply:
Made the condition clearer by explicitly using the undefined key ID as marker